### PR TITLE
MailingEventQueue - Use consistent CamelCaseName. Prevent conflicts.

### PR DESCRIPTION
**Pull request by totten**
Overview
---------

The entity `CRM_Mailing_Event_DAO_Queue` has some  inconsistencies in how the name is presented in its short/CamelCase form.

* The public APIv3 name presents as`MailingEventQueue`.
* The internal name reported by `AllCoreTables` is `Queue`.

The names should be consistent - keep the public-name `MailingEventQueue`  and change 1-2 internal spots that reference `Queue`.

Mismatched names create two issues:

* If we introduce `CRM_Queue_DAO_Queue`, it will cause a conflict within `AllCoreTables` (*two entities claiming to be `Queue`*).
* There is an asymptomatic miscalculation when processing APIv3 metadata for `MailingEventQueue` (mentioned below).

Unchanged
---------

The main/top-level names still appear the same for the following artifacts:

* XML File: `xml/schema/Mailing/Event/Queue.xml`
* DAO File: `CRM/Mailing/Event/DAO/Queue.php`
* BAO File: `CRM/Mailing/Event/BAO/Queue.php`
* APIv3 File: `api/v3/MailingEventQueue.php`
* APIv4 File: _Not applicable_
* Hook_pre/hook_post: _Not emitted (look at `CRM_Mailing_Event_BAO_Queue::create()` to confirm)_

Changed things: Before
---------------

* TLDR: Inconsistent mix of `MailingEventQueue` and `Queue`
* `AllCoreTables` lists the short/CamelCaseName as `Queue`, which
  does not match the short APIv3 name (`api/v3/MailingEventQueue.php`).
* When formatting results for `MailingEventQueue.create`, it
  uses `_civicrm_api3_basic_create(...'Queue')`. Again, that
  name does not match `api/v3/MailingEventQueue.php`.
  (Substantively, this means that it may call `Queue.getfields`
  to read extra metadata. But this returns an empty list...)

Changed things: After
---------------

* TLDR: Consistent use of `MailingEventQueue`
* `AllCoreTables` lists the short/CamleCaseName as `MailingEventQueue`
  which does match the `api/v3/MailingEventQueue.php`
* When formatting results for `MailingEventQueue.create`, it
  uses `_civicrm_api3_basic_create(...'MailingEventQueue')`.
  This matches. (Substantively, this means that it may call `MailingEventQueue.getfields`
  to read extra metadata. As before, this returns an empty list...)

Comments
----------------

Two major factors to consider re:QA:

* The patch touches `GenCode`. To ensure no surprise side-effects, I regenerated all DAOs (`GENCODE_FORCE=1 ./bin/setup.sh -g`). They matched up.
* The main purpose is to prevent conflicts with other DAOs/APIs - ie [new Queue API](https://github.com/civicrm/civicrm-core/pull/22324/commits/56140a0d577f55e628030d6130bfcb3d9d233463). There's a working demonstration in #22324 -- without this patch,it fails on conflicts in `SyntaxConformanceTest`. With the patch, it passes..

### CODE REVIEW

**Comment by totten** [comment]
@colemanw This appears to be the root cause of the naming-conflict we discussed before. With this fix, #22324's [Civi\Api4\Queue](https://github.com/civicrm/civicrm-core/pull/22324/commits/a6b66e6029fbe0b61c8381de3b4e98f469dc2fd8) passes the `ConformanceTest`.

**Comment by colemanw** [comment]
This is a good change and should be merged pending tests passing.
I'm curious @totten I see you had to add a new xml tag to allow overriding the short name. Is this really the first instance where that need has arisen? I seem to recall others, e.g. I think CiviDiscount has `CRM_Discount_DAO_Item` which resolves to the most ambiguous short name in the universe.

**Comment by totten** [comment]
> Is this really the first instance where that need has arisen? 

The closest I can think of is how APIv4 needs to use a different name for `Case` (`CRM_Case_BAO_Case` / `Civi\Api4\CiviCase`), but that's a fairly different set of goals+mechanics. (Ex: `Case`s have full remote CRUD API support. They also have pre-existing signatures for `hook_pre`/`hook_post`.)

From the perspective of how `GenCode` produces `AllCoreTables`, I couldn't find anything else right now. However, it would not be surprising to find latent issues when/if one starts adding APIs for more second-tier entities like `CRM_Discount_DAO_Item`...
